qpmodel/ExprFunc.cs

```diff
@@ -744,7 +744,7 @@ public override void Bind(BindContext context)
             base.Bind(context);
             type_ = then_()[0].type_;
             if (else_() != null)
-                Debug.Assert(type_.Compatible(else_().type_));
+                Debug.Assert(TypeBase.Compatible(type_, else_().type_));
         }
```

Comment:
This assert will always be TRUE (never fails) because TypeBase.Compatible() is returning true unconditionally. Do you intend to add actual compatibility checks in TypeBase.Compatible() later, or the actual compatibility is determined elsewhere?

Reply:
It is true because we haven't implemented this functionality. This functionality shall be done as part of #183

Reply:
Got it.